### PR TITLE
chore: Refactor how `user.role` are checked into methods

### PR DESCRIPTION
**Pull request by leobenkel**
# Pull Request Checklist

### Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

https://github.com/open-webui/open-webui/discussions/4316

**Before submitting, make sure you've checked the following:**

- [x] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [x] **Description:** Provide a concise description of the changes made in this pull request.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [x] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [x] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
- [x] **Testing:** Have you written and run sufficient tests for validating the changes?
- [x] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [x] **Prefix:** To cleary categorize this pull request, prefix the pull request title, using one of the following:
  - **BREAKING CHANGE**: Significant changes that may affect compatibility
  - **build**: Changes that affect the build system or external dependencies
  - **ci**: Changes to our continuous integration processes or workflows
  - **chore**: Refactor, cleanup, or other non-functional code changes
  - **docs**: Documentation update or addition
  - **feat**: Introduces a new feature or enhancement to the codebase
  - **fix**: Bug fix or error correction
  - **i18n**: Internationalization or localization changes
  - **perf**: Performance improvement
  - **refactor**: Code restructuring for better maintainability, readability, or scalability
  - **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)
  - **test**: Adding missing tests or correcting existing tests
  - **WIP**: Work in progress, a temporary label for incomplete or ongoing work

# Changelog Entry

### Description

- Refactor how `user.role` are checked into methods

### Added

- Refactor how `user.role` are checked into methods

### Changed

- Refactor how `user.role` are checked into methods

### CODE REVIEW

**Comment by leobenkel** [comment]
https://github.com/open-webui/open-webui/discussions/4316

File: backend/apps/webui/models/users.py [hunk 1]
```diff
@@ -59,6 +59,18 @@ class UserModel(BaseModel):
 
     model_config = ConfigDict(from_attributes=True)
 
+    def is_admin(self):
```
Comment (tjbck) [inline backend/apps/webui/models/users.py:62]: `has_admin_role`

File: backend/apps/webui/models/users.py [hunk 2]
```diff
@@ -59,6 +59,18 @@ class UserModel(BaseModel):
...
+        return self.role == "admin"
+
+    def is_user(self):
```
Comment (tjbck) [inline backend/apps/webui/models/users.py:65]: `has_user_role`

**Comment by tjbck** [comment]
Would be better to have Enums here, closing this PR.

**Comment by leobenkel** [comment]
We could iterate on this PR instead of closing it right away ....

**Comment by leobenkel** [comment]
by behaving that way, you are only going to push contributors away. just an advice.

**Pull request closed without merging**